src/sio.spec.ts

```diff
@@ -0,0 +1,82 @@
+import { RP2040, SIO_START_ADDRESS } from './rp2040';
+
+import { GDBClient } from './utils/gdbclient';
+import { ICortexTestDriver } from './utils/test-driver';
+import { GDBTestDriver } from './utils/test-driver-gdb';
+import { RP2040TestDriver } from './utils/test-driver-rp2040';
+
+//HARDWARE DIVIDER
+const SIO_DIV_UDIVIDEND = SIO_START_ADDRESS + 0x060; //  Divider unsigned dividend
+const SIO_DIV_UDIVISOR = SIO_START_ADDRESS + 0x064; //  Divider unsigned divisor
+const SIO_DIV_SDIVIDEND = SIO_START_ADDRESS + 0x068; //  Divider signed dividend
+const SIO_DIV_SDIVISOR = SIO_START_ADDRESS + 0x06c; //  Divider signed divisor
+const SIO_DIV_QUOTIENT = SIO_START_ADDRESS + 0x070; //  Divider result quotient
+const SIO_DIV_REMAINDER = SIO_START_ADDRESS + 0x074; //Divider result remainder
+const SIO_DIV_CSR = SIO_START_ADDRESS + 0x078;
+
+describe('RPSIO', () => {
+  let cpu: ICortexTestDriver;
+
+  beforeEach(async () => {
+    // if (process.env.TEST_GDB_SERVER) {
+    if (false) {
+      const client = new GDBClient();
+      await client.connect('127.0.0.1');
+      cpu = new GDBTestDriver(client);
+      await cpu.init();
+    } else {
+      cpu = new RP2040TestDriver(new RP2040());
+    }
+  });
+
+  afterEach(async () => {
+    await cpu.tearDown();
+  });
+
+  describe('Hardware Divider', () => {
+    it('should set perform a signed hardware divider 123456 / -321 = -384 R192', async () => {
```

Comment:
I think "REM 192" will be clearer than "R192" (same for the other test names).

Also, the word "set" (second word) seems not related here?